### PR TITLE
Narven/delete bug fix

### PR DESCRIPTION
**Pull request by NaraVen**
Version upgrade
Removed data comparison for token check for performance
Reenabled MSI to run under a separate thread.

### CODE REVIEW

File: blobfuse/src/OAuthTokenCredentialManager.cpp [hunk 1]
```diff
@@ -194,36 +185,14 @@ OAuthToken OAuthTokenCredentialManager::refresh_token()
...
-                // If we were working in Go, I (Adele) would suggest using a sync.WaitGroup for this functionality.
-                syslog(LOG_ERR, "OAUTH Token : Wait for token to get refreshed as buffer time has expired");
-                token_mutex.lock();
-            }
-        }
-
-        // Note that we don't always lock in this function and sometimes return early.
-        // Be sure to ensure you're safely manipulating the lock when modifying this function.
-        token_mutex.unlock();
+    #ifndef TOKEN_REFRESH_THREAD
```
Comment (vibhansa-msft) [inline blobfuse/src/OAuthTokenCredentialManager.cpp:188]: This check shall be reverse, we want to use previously acquired token if the monitor thread is running.

File: blobfuse/src/OAuthTokenCredentialManager.cpp [hunk 2]
```diff
@@ -194,36 +185,14 @@ OAuthToken OAuthTokenCredentialManager::refresh_token()
...
+    // since the thread would have updated the expired token just check if valid_authentication
+   // and return the current token, control will fall to the if is_token_expired() below if this directive is undefined.
+    if (valid_authentication)
+    {
+        return current_oauth_token;
     }
-    #else
+    #endif
```
Comment (vibhansa-msft) [inline blobfuse/src/OAuthTokenCredentialManager.cpp:195]: No need to call refresh_token() if the thread is running, we just need to wait for the thread to refresh the token.

Comment (NaraVen) [reply blobfuse/src/OAuthTokenCredentialManager.cpp:195]: if valid_authentication has failed the thread has gone wrong so refreshing the token here even if the #def TOKEN_REFRESH_THREAD isdefined

File: blobfuse/src/OAuthTokenCredentialManager.cpp (hunk 1 above)
Comment (NaraVen) [reply blobfuse/src/OAuthTokenCredentialManager.cpp]: done

File: blobfuse/src/OAuthTokenCredentialManager.cpp (hunk 2 above)
Comment (vibhansa-msft) [reply blobfuse/src/OAuthTokenCredentialManager.cpp:195]: understood the logic.

**Review by vibhansa-msft — APPROVED** [review]

**Pull request merged**